### PR TITLE
[AKS] Fix apiServerAccessProfile when updating --api-server-authorized-ip-ranges

### LINKED ISSUE #12587: 'az aks update --api-server-authorized-ip-ranges' will also enable private cluster
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 
2.2

**Describe the bug**
az aks update -g <>-n <> --api-server-authorized-ip-ranges '' --debug 

Will also update 'enablePrivateCluster' to true
"apiServerAccessProfile": {"authorizedIPRanges": [], "enablePrivateCluster": true}}}

This PR adds 2nd parameter instance to _populate_api_server_access_profile
https://github.com/Azure/azure-cli/pull/10976/files
This PR adds another parameter to _populate_api_server_access_profile, only updates aks_create call, didn't update aks_update call
https://github.com/Azure/azure-cli/pull/12353

**To Reproduce**
Run command above

**Expected behavior**
Do not touch enablePrivateCluster behaviour

**Environment summary**
Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)  

**Additional context**

### PR DESCRIPTION
**Pull request by feiskyer**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[AKS] Fix apiServerAccessProfile when updating --api-server-authorized-ip-ranges

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

Fixes #12587

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

**Comment by yungezz** [comment]
hi @feiskyer  could you pls fix the CI failure?

**Comment by feiskyer** [comment]
@yungezz The error message shows it needs 'az login' and I found there are a lot of credentials in test recordings files. Why do we need to commit the credentials in the opensource project?

**Comment by feiskyer** [comment]
Let's see whether the tests could pass without test recording files.

**Comment by feiskyer** [comment]
CI passed with faked credentials. PTAL @qwordy @yungezz

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -643,6 +643,208 @@ def test_aks_create_default_setting(self, resource_group, resource_group_locatio
         #delete
         self.cmd('aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus')
+    @RoleBasedServicePrincipalPreparer()
+    def test_aks_create_private_cluster_then_update(self, resource_group, resource_group_location, sp_name, sp_password):
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py]: nice test!

File: src/azure-cli/azure/cli/command_modules/acs/_helpers.py [hunk 2]
```diff
@@ -22,6 +23,9 @@ def _populate_api_server_access_profile(api_server_authorized_ip_ranges, enable_
     else:
         authorized_ip_ranges = [ip.strip() for ip in api_server_authorized_ip_ranges.split(",")]
 
+    if profile.enable_private_cluster and authorized_ip_ranges:
+        raise CLIError('authorized_ip_ranges is not supported for private cluster')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/_helpers.py]: --authorized-ip-ranges is better. authorized_ip_ranges is internal name

Comment (feiskyer) [reply src/azure-cli/azure/cli/command_modules/acs/_helpers.py]: fixed in the new commit

**Pull request merged**